src/SortByFieldExtension.php

```diff
@@ -84,11 +96,27 @@ public function sortByFieldFilter($content, $sort_by = null, $direction = 'asc')
      * @return bool If collection item can be sorted
      */
     private static function isSortable($item, $field) {
-        if (is_array($item))
-            return array_key_exists($field, $item);
-        elseif (is_object($item))
-            return isset($item->$field) || property_exists($item, $field);
-        else
-            return false;
+
+        $isSortable = false;
+
+        foreach (explode('.', $field) as $field) {
+            if (is_array($item)) {
+                $isSortable = array_key_exists($field, $item);
+                if ($isSortable) {
+                    $item = $item[$field];
+                }
+            } elseif (is_object($item)) {
+                $isSortable = isset($item->$field) ||
+                    method_exists($item, 'get'.ucfirst($field));
+                if ($isSortable) {
+                    $item = isset($item->$field)
```

Comment:
Nested ternary operator on multiple lines, should properly not be a ternary operator. This is not particularity readable.

Reply:
I`am pushed fixes